### DISCUSSION TITLE
System identifier variable for system metrics + Class Numbers for Confusion Matrix

### DISCUSSION THREAD

**Discussion by digantamisra98** [ContinualAI/avalanche#459] [Feature Request] [answered]
For the W&B Logger currently under WIP (see [447](https://github.com/ContinualAI/avalanche/pull/447)), it would be great to nest the system metrics under a super section in the dashboard called as "System" and the performance metrics under "Performance", for the same, it would be super useful to have this variable in all system metric classes like `self.system = True` so that I won't have to keep a list of hard-coded system metric names to put within the super section. Additionally, for an interactive Confusion Matrix logger, it would be required to have the class names/ class numbers as mentioned in the docs [here](https://docs.wandb.ai/library/log#custom-charts).

**Comment by AndreaCossu** [comment, marked as answer]
I think we can provide all of that. I'm going to work on it :smile:

**Reply by digantamisra98** [reply]
Thanks, looking forward to the updates!

**Reply by AndreaCossu** [reply]
Hi @digantamisra98. I gave some thoughts on your requests. I think that the `system` parameter within the metric would not help you because the logger receives a `MetricValue` (name, y value, x value), not the metric class instance. I would avoid to add a new value within `MetricValue`. I think for the moment you can keep a list of performance metrics names and check if the incoming metric name is in the list.

**Reply by AndreaCossu** [reply]
As for the Confusion Matrix, I think we have the same issue here. We could provide two different confusion matrixes implementations, but that is not ideal in my opinion. What if, for the moment, we stick with logging the image produced by the current confusion matrix implementation? We can think about implementing the W&B one later. Let me know what you think :smile:

**Reply by digantamisra98** [reply]
Understandable for metrics, I can maintain a list, however, dynamic and interactive logging for Confusion Matrix seems to be crucial since it has features which can be useful to many. Is there anyway, we can pass the instance to the logger? Unless it's not possible at all, we can settle for image based confusion matrix logging which is already working as per the PR.

**Reply by AndreaCossu** [reply]
I think we can provide this implementation for the moment. I'd like to discuss the option of implementing a separate Confusion Matrix for W&B, so that the user can log both the image and the W&B empowered one. That should be easy. We can discuss it during the today's meeting. If you can confirm that the PR code is working I'll merge the PR as soon as it passes the tests.

**Reply by digantamisra98** [reply]
Sure, I'll pass the PR tests in a while. Looking forward to today's meeting.
